File: CustomerApp/Models/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace CustomerApp.Models
{
    public class Customer
    {
        [Key]
        public int Id { get; set; }
        [Required]

        [Display(Name = "First Name")]
        public string? FirstName { get; set; }

        [Required(ErrorMessage = "Last Name is required.")]
        [Display(Name = "Last Name")]
        public string? LastName { get; set; }

        [Required(ErrorMessage = "Email address is required.")]
        [EmailAddress(ErrorMessage = "Invalid email address.")]
        public string? Email { get; set; }

        public DateTime Created { get; set; } 
        public DateTime Updated { get; set; } = DateTime.Now;
    }
}

File: CustomerApp/ClientApp/src/app/models/customer.ts
export interface Customer {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  created?: Date;
  updated?: Date;
  isEdit: boolean;
}

export const CustomerColumns = [
 
  {
    key: 'id',
    type: 'number',
    label: 'Customer Id',
    isEditable: false,
  },
  {
    key: 'firstName',
    type: 'text',
    label: 'First Name',
    isEditable: true,
  },
  {
    key: 'lastName',
    type: 'text',
    label: 'Last Name',
    isEditable: true,
  },
  {
    key: 'email',
    type: 'email',
    label: 'Email',
    isEditable: true,
  },
  {
    key: 'created',
    type: 'date',
    label: 'Created',
    isEditable: false,
  },
  {
    key: 'updated',
    type: 'date',
    label: 'Updated',
    isEditable: false,
  },
  {
    key: 'isEdit',
    type: 'isEdit',
    label: '',
    isEditable: true,
  },
];
